time/src/serde/mod.rs

```diff
@@ -252,10 +252,18 @@ impl Serialize for Duration {
     fn serialize<S: Serializer>(&self, serializer: S) -> Result<S::Ok, S::Error> {
         #[cfg(feature = "serde-human-readable")]
         if serializer.is_human_readable() {
+            // Simply extracting sign and then calling `self.whole_seconds().abs()` will
+            // cause overflow on `i64::MIN` seconds, hence this kludge.
+            let sign_fix = if self.is_negative() && self.whole_seconds() == 0 {
+                "-"
+            } else {
+                ""
+            };
+
             return serializer.collect_str(&format_args!(
-                "{}.{:>09}",
-                self.whole_seconds(),
-                self.subsec_nanoseconds().abs()
+                "{sign_fix}{seconds}.{nanoseconds:>09}",
+                seconds = self.whole_seconds(),
```

Comment:
There's no need to name these, as they're used in the order they're provided.

Reply:
The point of named format args is not only to allow reordering, but also making the format string more readable (and less mistake-prone).

Reply:
Reordering isn't desirable here, as they _should_ be in that order. I'd prefer leaving them as-is (i.e. without explicit names). It's standard practice in my experience.